### PR TITLE
GalleryRP 2023-07-03 release 2nd

### PR DESCRIPTION
**Pull request by Sandido**
# Contributing to the Azure SDK

Based on this PR: https://github.com/azure-sdk/azure-sdk-for-net/pull/6805/files
From this swagger release: https://github.com/Azure/azure-rest-api-specs/pull/27518

This branch is a copy of this one, https://github.com/Azure/azure-sdk-for-net/pull/41766 , trying to resolve the build error stating the code is out of date. 

Please see our [CONTRIBUTING.md](https://github.com/Azure/azure-sdk-for-net/blob/main/CONTRIBUTING.md) if you are not familiar with contributing to this repository or have questions.

For specific information about pull request etiquette and best practices, see [this section](https://github.com/Azure/azure-sdk-for-net/blob/main/CONTRIBUTING.md#pull-request-etiquette-and-best-practices).

### CODE REVIEW

**Comment by Sandido** [comment]
Rebased this branch to another that is up to date with main, seeing if that resolves the build error stating the code is out of date.

File: sdk/compute/Azure.ResourceManager.Compute/api/Azure.ResourceManager.Compute.netstandard2.0.cs [hunk 1]
```diff
@@ -4934,6 +4934,7 @@ public partial class GalleryArtifactVersionFullSource : Azure.ResourceManager.Co
         public string CommunityGalleryImageId { get { throw null; } set { } }
         [System.ComponentModel.EditorBrowsableAttribute(System.ComponentModel.EditorBrowsableState.Never)]
         public override System.Uri Uri { get { throw null; } set { } }
+        public string VirtualMachineId { get { throw null; } set { } }
```
Comment (ArcturusZhang) [inline sdk/compute/Azure.ResourceManager.Compute/api/Azure.ResourceManager.Compute.netstandard2.0.cs:4937]: by its description, this is an arm resource id.
Could we change the format of this property to `arm-id`?

**Review by ArcturusZhang — COMMENTED** [review]
Please update the Changelog.md accordingly with your changes.

File: sdk/compute/Azure.ResourceManager.Compute/api/Azure.ResourceManager.Compute.netstandard2.0.cs (hunk 1 above)
Comment (ArcturusZhang) [reply sdk/compute/Azure.ResourceManager.Compute/api/Azure.ResourceManager.Compute.netstandard2.0.cs:4937]: You could do this by adding:
```
GalleryArtifactVersionFullSource.virtualMachineId: -|arm-id
```
to the `rename-mapping` list in `autorest.md`, then please regen the code and API file.

**Comment by Sandido** [comment]
@ArcturusZhang , edits made, please re-review!

File: sdk/compute/Azure.ResourceManager.Compute/CHANGELOG.md [hunk 2]
```diff
@@ -1,14 +1,11 @@
 # Release History
 
-## 1.4.0-beta.2 (Unreleased)
+## 1.4.0 (2024-02-21)
```
[the remark this answers is not in the document]
Comment (Sandido) [reply sdk/compute/Azure.ResourceManager.Compute/CHANGELOG.md:3]: ##[error]ChangeLog[D:\a\_work\1\s\sdk\compute\Azure.ResourceManager.Compute\CHANGELOG.md] does not have an entry for version 1.4.0-beta.2.

**Review by ArcturusZhang — APPROVED** [review]

**Pull request merged**